packages/core/src/client.ts

```diff
@@ -2206,6 +2206,11 @@ export class MedplumClient extends EventTarget {
       return undefined as unknown as T;
     }
 
+    if (response.status === 404) {
```

Comment:
It looks like we might be kind of playing whack-a-mole with error cases here; would it be better to handle anything `response.status` that ISN'T 200/201/202 by outputting an error message?

Reply:
@mattwiller to your point, it seems like we do handle all responses >= 400 on line 2222.

@codyebberson   - do you remember why we handle 400's after the `.json()` parse?

Reply:
Yeah, I don't think this is right.

We parse the body because it often includes more details about the error.  This *should* have been handled by the check below:

```ts
    if (response.status >= 400) {
      throw new OperationOutcomeError(normalizeOperationOutcome(obj));
    }
```

I don't know why that code path didn't solve this particular case?

Reply:
@codyebberson

FYI that error handling code is problematic depending on the FHIR server you're using.

For example with HAPI, their error responses are sometimes XML if `response.status >= 400`, in which case normalizing the operation outcome throws an error - since this always expects a JSON response. Then, we're left with a JSON parsing error, and lose the original context of the server response.

We'll raise a separate issue for this, and/or push a PR for the client to fix this.